src/features/post/model/usePostDetailQuery.ts

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from '@tanstack/react-query';
+import { getPostDetail } from '../api/getPostDetail';
+import { transformDetailToPost } from '@/entities/post/model/mappers';
+import { PostDetail } from '@/entities/post/model/types';
+
+/**
+ * 게시글 상세 조회 React Query 훅
+ */
+export const usePostDetail = (postId: number) =>
+  useQuery<PostDetail>({
+    queryKey: ['postDetail', postId],
+    queryFn: async () => {
+      const detail = await getPostDetail(postId);
+      return transformDetailToPost(detail);
+    },
+    enabled: postId != null,
+  });
```

Comment:
```suggestion
export const usePostDetail = (postId: number, options?: { enabled?: boolean }) =>
  useQuery<PostDetail>({
    queryKey: ['postDetail', postId],
    queryFn: async () => {
      const detail = await getPostDetail(postId);
      return transformDetailToPost(detail);
    },
    enabled: options?.enabled ?? postId != null,
  });
```
혹시 이렇게 options props 추가해도 될까요?
게시글 생성/수정 구현에서 생성 모드면 이 훅을 호출하면 안되는데, postId를 null로 놓으면 타입 에러가 나서요!

Reply:
네 그렇게 추가해두겠습니다!